File: BackEnd/proyecto/src/main/java/com/example/proyecto/model/Falta.java
package com.example.proyecto.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "faltas")
public class Falta {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @Column(name = "hora", nullable = false)
    private Integer hora;

    @Column(name = "fecha")
    private LocalDate fecha;

    @Column(name = "justificada")
    private boolean justificada;

    @Column(name = "id_Alumno")
    private int idAlumno;

}

File: FrontEnd/app/src/main/java/com/example/tycep_fe/models/Falta.kt
package com.example.tycep_fe.models

import com.google.gson.annotations.SerializedName
import java.lang.AssertionError
import java.time.LocalDate

data class Falta(
    @SerializedName("id")
    val id:Int?,
    @SerializedName("hora")
    val hora:Int,
    @SerializedName("asignatura")
    val asignatura: String,
    @SerializedName("fecha")
    val fecha:String,
    @SerializedName("idAlumno")
    val idAlumno:Int,
    @SerializedName("justificada")
    val justificada:Boolean
)
